src/experiments.js

```diff
@@ -209,15 +206,59 @@ Test.goalHandlers = {};
  * @param <Function> callback to bind an event listener.
  * @param <String[]> array of allowed node types to bind listener to.
  */
-Test.registerGoal = ( name, callback, closest = [] ) => {
-	Test.goalHandlers[ name ] = {
+const registerGoalHandler = ( name, callback, closest = [] ) => {
+	goalHandlers[ name ] = {
 		callback,
 		closest: Array.isArray( closest ) ? closest : [ closest ],
 	};
 };
 
+/**
+ *
+ * @param {HTMLElement} node The target node to attach the event listener to.
+ * @param {*} record The function called by the event listener to record an event.
+ * @param {*} on The JS dvent name to listen on.
+ */
+const bindListener = ( node, record, on ) => node && node.addEventListener( on, event => {
+	typeof record === 'function' && record( event );
+} );
+
+/**
+ * Get a goal handling function.
+ *
+ * @param {String} name The name of a registered goal, goal handler or a valid JS event.
+ * @param {Object} options Optional overrides for the registered goal properties.
+ */
+const getGoalHandler = ( name, options = {} ) => {
+	// Compile the goal configuration.
+	const goal = Object.assign(
+		{
+			name,
+			event: name,
+			callback: bindListener,
+		},
+		window.Altis.Analytics.Experiments.Goals[ name ] || {},
+		goalHandlers[ name ] || {},
+		options
+	);
```

Comment:
Can this be replaced with a spread instead?
```suggestion
	const goal = {
		name,
		event: name,
		callback: bindListener,
		...( window.Altis.Analytics.Experiments.Goals[ name ] || {} ),
		...( goalHandlers[ name ] || {} ),
		...options
	};
```

Reply:
Would it actually compile to the same or could it compile to something larger? I don't really know what the benefit of doing this would be.

Reply:
It'll compile to the same in the newest browsers, but spreads are clearer that there's no mutation; Object.assign mutates the initial parameter.